### DISCUSSION TITLE
Treatlife Smart Ceiling Fan Control and Light Dimmer Switch

### DISCUSSION THREAD

**Discussion by TubadoG** [milo526/homebridge-tuya-web#222] [Q&A]
Team, I'm looking to use the Treatlife Smart Ceiling Fan Control and Light Dimmer Switch in HomeKit.  I was able to connect to my switch without problems, but in the HimeKit app, I can only see the Ceiling Fan, I can't see the Light Switch.  I'm curious if there are any thoughts or workarounds for this since it is in essence, two smart devices in one.

As a side note, both switches work as expected in the SmartLife (Tuya) app.  

![image](https://user-images.githubusercontent.com/5658994/110179076-75a22a00-7dd5-11eb-8d8d-8688d444c082.png)

![image](https://user-images.githubusercontent.com/5658994/110179131-8b175400-7dd5-11eb-82d2-9dd79f5282ad.png)

**Comment by milo526** [comment]
Sadly the API that this plugin uses is severely limited in this regard.
The plugin can't get any information on the light in this device.

You could solve this partially by creating scenes in Tuya and exposing them. 
Create 1 scene for on and 1 for off, expose these through the plugin.

**Reply by TubadoG** [reply]
Thanks for that note. Your solution makes sense, but as you know, it’s not ideal.  I’m thinking that I’ll need to flash Tasmota onto the switch, which I should do anyway to get this out of the cloud.  I’ll try some HTML sniffing first before I change lanes.  Thanks again!
